### ISSUE TITLE
Bug? colloquialSinceNow

### ISSUE THREAD

**Issue by ShockUtility** [malcommac/SwiftDate#417]
XCode : 8.3.1
SwiftDate : 4.1.1

I did two tests.
The second thing is wrong.
Is the bug correct?

```swift
// Test 1
if let date = "2017-04-15 12:46:20 +0000".date(format: .iso8601Auto) {
  print("Today  : ",Date())
  print("Source : ",date.absoluteDate)
  print(try! date.colloquialSinceNow())
}
```
> Today  :  2017-04-15 14:54:32 +0000
> Source :  2017-04-15 12:46:20 +0000
> (colloquial: "2 hours ago", time: Optional("at 12:46"))
        
```swift
// Test 2
if let date = "2017-04-13 12:09:04 +0000".date(format: .iso8601Auto) {
  print("Today  : ",Date())
  print("Source : ",date.absoluteDate)
  print(try! date.colloquialSinceNow())
}
```

> Today  :  2017-04-15 14:54:32 +0000
> Source :  2017-04-13 12:09:04 +0000
> (colloquial: "2 hours ago", time: Optional("at 12:09"))

There seems to be a problem with the following items.
--- DateInRegionFormatter.swift  (Line 220) ---
> "if cmp.hour == nil {"   =>    "if cmp.day! > 0 {"

**Comment by RolandasRazma** [comment]
it looks like only hour is returned un purpose, but definitely unexpected...

```swift
if cmp.day != 0 || !hasLowerAllowedComponents(than: .day) {
    if cmp.hour == nil {
        ...
    } else {
        // less than 48 hours ago
        ...
        let colloquial_date = try self.localized(unit: .hour, withValue: cmp.hour!, asFuture: isFuture, args: abs(cmp.hour!))
        return ...
    }
}
```

**Comment by RolandasRazma** [comment]
same as https://github.com/malcommac/SwiftDate/issues/419

**Comment by paulz** [comment]
We have an issue with colloquialSinceNow expressed in this spec:
```
                let threeDaysAgo = DateInRegion() - 3.day
                expect(try! threeDaysAgo.colloquialSinceNow().colloquial) == "3 days ago"
```
results in test failure:
```
expected to equal <3 days ago>, got <now>
```

**Comment by malcommac** [comment]
Just fixed in 4.1.2 alongside new unit tests.
Thank you!

**Comment by paulz** [comment]
Thank you, works for us!
